### DISCUSSION TITLE
延迟加载单个字段 怎么实现的？

### DISCUSSION THREAD

**Discussion by sgf** [dotnetcore/FreeSql#1709] [Q&A] [answered]
延迟加载单个字段 怎么实现的？ 某个字段体积较大，但是该字段仅在需要访问的时候，单独从数据库查询即可。
而在分页的时候，我会手动排除该字段的绑定，因此不会访问该字段，所以分页的时候就不会加载。

**Comment by 2881099** [comment]
这种需求，即使支持了可配置化，使用者配置的复杂度，可能已远超直接使用 ISelect 查询。

意义不大。

**Reply by sgf** [reply]
Select可以排除字段么。虽然可以选择字段，但是，每次Model更新的话，你懂的，总不见得每个地方都去，添加新的字段，但是如果可以排除特定字段的话，那就方便了。

**Reply by 2881099** [reply, marked as answer]
https://github.com/dotnetcore/FreeSql/issues/528

**Reply by sgf** [reply]
配置肯定应该是要做的很简单的,只是不太好实现吧. 字段直接加Lazy包一下不就好了,Lazy.Value的时候起一个请求去查询,当然这样的话,Entity本身可能需要做静态编织的包装.

**Reply by sgf** [reply]
编码时:
```C#
    class Entiry
    {
        Lazy<byte[]> Data { get; set; }
    }
```
编译后:
```C#
    class Entiry
    {
        Lazy<byte[]> Data
        {
            get => _freeSql.Select<Entiry>().Where(x => x.Id = this.ID).ToList(x => new { x.Id, x.Data });
            set => _freeSql.Update<Entiry>(this.ID).Set(a => a.Data, value).ExecuteAffrows();
        }
    }
```

**Reply by sgf** [reply]
我这个只是随便一写.实际应该可以完善.
